presto-cache/src/main/java/com/facebook/presto/cache/filemerge/FileMergeCacheManager.java

```diff
@@ -85,7 +89,8 @@
     // a mapping from remote file `F` to a range map `M`; the corresponding local cache file for each range in `M` represents the cached chunk of `F`
     private final Map<Path, CacheRange> persistedRanges = new ConcurrentHashMap<>();
     // a local cache only to control the lifecycle of persisted
-    private final Cache<Path, Boolean> cache;
+    private final Cache<Path, String> cache;
+    private final Map<String, Set<Path>> tableFiles = new ConcurrentHashMap<>();
```

Comment:
`private final Map<Long, Set<Path>> scopeFiles = new ConcurrentHashMap<>();`, where `Long` is gotten from `CacheScopeHandle.getIdentifier()`